config/puma/production.rb

```diff
@@ -27,6 +27,22 @@
 #
 preload_app!
 
+# Check if FOREMAN_BIND was set to an IP address based on the previous
+# definition of FOREMAN_BIND. If it is an IP address, define the host
+# and port through Puma's DSL. Otherwise rescue and assume the new
+# style of a fully specified bind in the formats of:
+#
+#  * unix:///run/foreman.sock
+#  * tcp://127.0.0.1:3000
+#
+begin
+  IPAddr.new(ENV['FOREMAN_BIND'])
+
+  port ENV.fetch('FOREMAN_PORT', '3000'), ENV['FOREMAN_BIND']
+rescue IPAddr::Error
+  bind ENV.fetch('FOREMAN_BIND', 'tcp://127.0.0.1:3000')
```

Comment:
Note this changes the previous default of 0.0.0.0 to 127.0.0.1. That breaks deployments without Apache. I don't mind if it binds to that by default here, but in `foreman.service` I would have an explicit `FOREMAN_BIND`. An additional benefit is that `systemctl cat foreman.service` will tell you how it's binding. That also matches this:

https://github.com/theforeman/foreman/blob/fc2387648abc558d8f3a8d9f4d914530f5823587/extras/systemd/foreman.socket#L9

Having this match will mean `yum install foreman-service` will have a working setup.